engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ExecuteJob.scala

```diff
@@ -27,63 +27,69 @@ class ExecuteJob(
 
   /** @inheritdoc */
   override def run(implicit ctx: RuntimeContext): Unit = {
+    try {
+      runImpl
+    } catch {
+      case t: Throwable =>
+        ctx.endpoint.sendToClient(
+          Api.Response(
+            Api.ExecutionFailed(
+              contextId,
+              Api.ExecutionResult.Failure(t.getMessage, None)
+            )
+          )
+        )
+    }
+  }
+
+  private def runImpl(implicit ctx: RuntimeContext): Unit = {
+    ctx.state.executionHooks.run()
```

Comment:
I'm thinking of wrapping it in a write compilation lock because hooks are supposed to change the context state

Reply:
And then we can remove the locking logic from individual hooks